synapse/lib/stormlib/model.py

```diff
@@ -74,6 +75,91 @@
+        ),
+        'storm': '''
+            init {
+                if $cmdopts.unlock {
+                    $lib.print("Unlocking: {name}", name=$cmdopts.name)
+                    $lib.model.deprecated.lock($cmdopts.name, $lib.false)
+                } else {
+                    if ($cmdopts.name = "*") {
+                        $lib.print("Locking all deprecated model elements.")
+                        for ($name, $locked) in $lib.model.deprecated.locks() {
+                            if (not $locked) { $lib.model.deprecated.lock($name, $lib.true) }
+                        }
+                    } else {
+                        $lib.print("Locking: {name}", name=$cmdopts.name)
+                        $lib.model.deprecated.lock($cmdopts.name, $lib.true)
+                    }
+                }
+            }
+        ''',
+    },
+    {
+        'name': 'model.deprecated.locks',
+        'descr': 'Display lock status of deprecated model elements.',
+        'storm': '''
+            $locks = $lib.model.deprecated.locks()
+            if $locks {
+                $lib.print("Lock status for deprecated forms/props:")
+                for ($name, $locked) in $lib.sorted($locks) {
+                    $lib.print("{name}: {locked}", name=$name, locked=$locked)
+                }
+            } else {
+                $lib.print("No deprecated locks found.")
+            }
+        ''',
+    },
+    {
+        'name': 'model.deprecated.check',
+        'descr': 'Check for lock status and the existance of deprecated model elements',
+        'storm': '''
+            init {
+
+                $ok = $lib.true
+                $lib.print("Checking the cortex for 3.0.0 upgrade readiness...")
+
+                $locks = $lib.model.deprecated.locks()
+
+                $lib.print("Checking deprecated model locks:")
+                for ($name, $locked) in $locks {
+                    if $locked {
+                        $lib.print("{name} is locked", name=$name)
+                    } else {
+                        $lib.warn("{name} is not yet locked", name=$name)
+                        $ok = $lib.false
+                    }
+
+                }
+
+                $lib.print("Checking for existance of deprecated model elements:")
+                for ($name, $locked) in $locks {
+
+                    $lib.print("{name}...", name=$name)
+
+                    for $layr in $lib.layer.list() {
+                        if $layr.getPropCount($name, maxsize=1) {
+                            $lib.warn("Layer {iden} still contains {name}", iden=$layr.iden, name=$name)
+                            $ok = $lib.false
+                        }
+                    }
+                }
+
+                if (not $ok) {
+                    $lib.print("Whups! Your Cortex needs some tweaks to be fully future-model compliant.")
```

Comment:
`Your Cortex currently has deprecated model elements in it.` or something. This message is unclear how to fix the problem or give hints towards how to fix it.